### PR TITLE
Feature/3623 submission flow positive test result with/without given user consent

### PR DESCRIPTION
**Pull request by puneetmahali1**
## Description
This PR is containing:
- New Thank You Screen: Now possibility to enter the symptoms form there or cancel it and then land into the home screen 
  with  old Thank you Screen.
- Show Positive Test Result Screen: Now shows different content based on submission consent is given or not.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-3623

## Screenshot
K1
K2
K3
K4
P1
P2

### CODE REVIEW

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 1]
```diff
@@ -375,14 +389,8 @@ extension ExposureSubmissionCoordinator {
 	func showWarnOthersScreen() {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:389]: please check if this func might be private

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 2]
```diff
@@ -409,13 +417,90 @@ extension ExposureSubmissionCoordinator {
 		navigationController?.present(alert, animated: true, completion: nil)
 	}
 
+	
 	func showThankYouScreen() {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:421]:
```suggestion
	
	privet func showThankYouScreen() {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 3]
```diff
@@ -409,13 +417,90 @@ extension ExposureSubmissionCoordinator {
...
+		}
 
-	// MARK: - Private
+		push(thankYouVC)
 
+	}
+	
+	// MARK: - Internal
+	
+	func presentPositiveTestResultCancelAlert() {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:436]: is internal here correct?
```suggestion
	private func presentPositiveTestResultCancelAlert()
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 4]
```diff
@@ -409,13 +417,90 @@ extension ExposureSubmissionCoordinator {
...
+		
+		alert.addAction(UIAlertAction(
+							title: alertButtonRight,
+							style: .default)
+		)
+		navigationController.present(alert, animated: true, completion: nil)
+	}
+	
+	
+	func presentThankYouCancelAlert() {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:472]:
```suggestion
	private func presentThankYouCancelAlert() {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionIntroViewModel.swift [hunk 5]
```diff
@@ -13,18 +13,21 @@ class ExposureSubmissionIntroViewModel {
...
 		self.onHotlineButtonTap = onHotlineButtonTap
+		self.onThankYouScreen = onThankYouScreen
 	}
 	
 	// MARK: - Internal
 	
 	let onQRCodeButtonTap: () -> Void
 	let onTANButtonTap: () -> Void
 	let onHotlineButtonTap: () -> Void
+	let onThankYouScreen: () -> Void
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionIntroViewModel.swift:30]: I can't see this one get's used? If it is useless, please remove

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift [hunk 6]
```diff
@@ -206,21 +237,54 @@ class ExposureSubmissionTestResultViewModel {
...
+						iconTint: .enaColor(for: .riskHigh),
+						hairline: .iconAttached,
+						bottomSpacing: .normal
+					),
+					ExposureSubmissionDynamicCell.stepCell(
+						style: .body,
+						title: AppStrings.ExposureSubmissionPositiveTestResult.noConsentInfo3,
+						icon: UIImage(named: "Icons - Home"),
+						iconTint: .enaColor(for: .riskHigh),
+						hairline: .iconAttached,
```
[the remark this answers is not in the document]
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift]: Eagle eye 😄 It is removed.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 3 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:436]: Let me check...

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 4 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:472]: It seems, that the whole coordinator has some inconsistencies with visibility states. Let me check the whole thing.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 7]
```diff
@@ -409,13 +417,90 @@ extension ExposureSubmissionCoordinator {
...
+			AppStrings.ExposureSubmissionPositiveTestResult.noConsentAlertButtonRight
+		
+		let alert = UIAlertController(
+			title: alertTitle,
+			message: alertMessage,
+			preferredStyle: .alert)
+		
+		alert.addAction(UIAlertAction(
+							title: alertButtonLeft,
+							style: .cancel,
```
[the remark this answers is not in the document]
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: Good point. Changed it 👍

File: src/xcode/ENA/ENA/Resources/Assets/Assets.xcassets/Common/Icons - Lock.imageset/Contents.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+      "filename" : "Icon_Testergebnis_positiv_Lock_Identity_light.pdf",
+      "idiom" : "universal",
+      "scale" : "1x"
+    }
+  ],
+  "info" : {
+    "author" : "xcode",
+    "version" : 1
+  }
+}
```
[the remark this answers is not in the document]
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Resources/Assets/Assets.xcassets/Common/Icons - Lock.imageset/Contents.json]: I changed the icon to a universal black template lock image

File: src/xcode/ENA/ENA/Source/Scenes/NavigationControllerWithFooter/ENANavigationFooterItem.swift [hunk 9]
```diff
@@ -17,4 +17,5 @@ class ENANavigationFooterItem: UINavigationItem {
 	@IBInspectable dynamic var isSecondaryButtonLoading: Bool = false
 	@IBInspectable dynamic var secondaryButtonTitle: String = ""
 	@IBInspectable dynamic var secondaryButtonHasBorder: Bool = false
+	@IBInspectable dynamic var secondaryButtonHasBackground: Bool = false
```
[the remark this answers is not in the document]
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/NavigationControllerWithFooter/ENANavigationFooterItem.swift:20]: Thanks. Added the observation.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 10]
```diff
@@ -130,7 +136,8 @@ extension ExposureSubmissionCoordinator {
 		let viewModel = ExposureSubmissionIntroViewModel(
 			onQRCodeButtonTap: { [weak self] in self?.showQRInfoScreen() },
 			onTANButtonTap: { [weak self] in self?.showTanScreen() },
-			onHotlineButtonTap: { [weak self] in self?.showHotlineScreen() }
+			onHotlineButtonTap: { [weak self] in self?.showHotlineScreen() },
+			onThankYouScreen: { self.showThankYouScreen() }
```
[the remark this answers is not in the document]
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: Thanks Kai, It's removed.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionIntroViewModel.swift (hunk 5 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionIntroViewModel.swift]: Removed.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/ExposureSubmissionTestResultViewModelTests.swift [hunk 11]
```diff
@@ -362,7 +356,7 @@ class ExposureSubmissionTestResultViewModelTests: XCTestCase {
 			waitForExpectations(timeout: .short)
 		}
 	}
-	
+	/*
```
[the remark this answers is not in the document]
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/ExposureSubmissionTestResultViewModelTests.swift:359]: 100% ;-) Did forget to code cleanup the out commented code.  Removed 👍

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 12]
```diff
@@ -65,14 +67,27 @@ class ExposureSubmissionCoordinator: NSObject, ExposureSubmissionCoordinating, R
...
 		)
 	}
+	
+	// MARK: - Private
+	
+	private var isSubmissionConsentGiven: Bool = false
+	
+	private var cancellables: Set<AnyCancellable> = []
+	
+	private let exposureSubmissionService: ExposureSubmissionService
```
[the remark this answers is not in the document]
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:89]: It's used for UITests and I also use it to distinguish when displaying the alert box triggered by the coordinator.

[the remark this answers is not in the document]
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: Good point. Changed code to access information via the coordinator model

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 1 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: Refactored to private 👍

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 2 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: Refactored to private 👍

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 3 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: Refactored to private 👍

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 4 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: Refactored to private 👍

**Review by nickguendling — APPROVED** [review]

**Review by KaiTeuber — APPROVED** [review]

**Pull request merged**